core/src/main/scala/kafka/network/SocketServer.scala

```diff
@@ -120,7 +120,6 @@ class SocketServer(val config: KafkaConfig,
    */
   def startup(startProcessingRequests: Boolean = true): Unit = {
     this.synchronized {
-      connectionQuotas = new ConnectionQuotas(config, time, metrics)
```

Comment:
Changes for moving ConnectionQuotas to QuotaFactory would be in the second PR that has quota API related changes, right? In any case, notice that `SocketServer.shutdown` now also calls connectionQuotas.close(), so would need to move closing connectionQuotas to QuotaFactory as well. Adding `ConnectionQuotas.close` method was pretty recent change, perhaps before you started coding this. (not sure if you pulled the most recent kafka code).

Reply:
in retrospect, moving ConnectionQuotas to QuotaFactory is not necessary for either the ZK PR, or for the quotas API PR. I did it to make the `IpConfigHandler` more idiomatic, but it increases the blast radius of the code and isn't really necessary. I will revert that change for the AK PR.